### PR TITLE
Airflow orchestration of Dataproc Templates

### LINKED ISSUE #288: [Spike][Airflow] Explore on developing samples showing template execution using airflow
Figure out if airflow sample jobs can be created to show template launch from Cloud Composer / Airflow

### PR DESCRIPTION
**Pull request by NiloFreitas**
This PR shows how to trigger Dataproc Template templates using Airflow.
The use chooses which template they want to run, and specify its arguments using the .ini configuration file.
Then, the created DAG will use the BashOperator to trigger the Dataproc Template start.sh with the provided configuration.

Closes #288

### CODE REVIEW

File: python/bin/start.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ GCS_STAGING_LOCATION=`echo $GCS_STAGING_LOCATION | sed 's/\/*$//'`
 # Do not rebuild when SKIP_BUILD is specified
 # Usage: export SKIP_BUILD=true
 if [ -z "$SKIP_BUILD" ]; then
-    python ${PROJECT_ROOT_DIR}/setup.py bdist_egg --output=$PACKAGE_EGG_FILE
+    python3 ${PROJECT_ROOT_DIR}/setup.py bdist_egg --output=$PACKAGE_EGG_FILE
```
Comment (shashank-google) [inline python/bin/start.sh:36]: this feels like unexpected change. Is it right thing to do ?

File: airflow/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Airflow Orchestration of Dataproc Templates
```
Comment (shashank-google) [inline airflow/README.md:1]: For Airflow newbee like me, I need more instructions of what needs to be done.

File: airflow/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Content in this folder demonstrates how to run Dataproc Templates from DAGs running in Airflow
+
+1) Structure your Airflow dags folder following this structure:
+```
+dags/
+    |_submit_dataproc_template_serverless.py
+    |_config/
+            |_submit_dataproc_template_serverless.ini
+    |_dependencies/
+            |_dataproc_templates/
```
Comment (shashank-google) [inline airflow/README.md:12]: Structure definition is easily understandable.
But how does someone move `_dataproc_templates/` and it's subdirectories inside dependencies folder ?
In other words how to deploy this structure ?

File: airflow/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+your.template.third.argument=<value>
+your.template.fourth.argument=<value>
+```
+
+3) Run the unit tests:
+```
+pytest airflow/pyspark
+```
+
+4) See the DAG in Airflow with id: **submit_dataproc_template_serverless**
```
Comment (shashank-google) [inline airflow/README.md:36]: Is some command or details missing here.
How does `DAG in Airflow` will get deployed ? 
What steps need to be performed for getting to DAG, do someone need to run `_submit_dataproc_template_serverless.py`

File: airflow/README.md (hunk 3 above)
Comment (shashank-google) [reply airflow/README.md:12]: Aah I see you have included some `cp -r` instructions for airflow.
Maybe we can be a bit more descriptive here as well for non-composer deployment

File: airflow/pyspark/dags/config/submit_dataproc_template_serverless.ini [hunk 5]
```diff
@@ -0,0 +1,33 @@
+[COMPOSER]
+DAG_NAME=submit_dataproc_template_serverless
+SCHEDULE_INTERVAL=@hourly
```
Comment (shashank-google) [inline airflow/pyspark/dags/config/submit_dataproc_template_serverless.ini:3]: Should we add capability to run it one-time as well, without any scheduling like `SCHEDULEING=true / false`

**Comment by shashank-google** [comment]
Nicely done @NiloFreitas I have added some comments.
@ppaglilla @surjits254 can you please take a look at this as well.

File: python/bin/start.sh (hunk 1 above)
Comment (NiloFreitas) [reply python/bin/start.sh:36]: It should have been this way from the start.
Will run integration tests before merging.

File: airflow/README.md (hunk 3 above)
Comment (NiloFreitas) [reply airflow/README.md:14]: Added deployment instructions

File: airflow/README.md (hunk 4 above)
Comment (NiloFreitas) [reply airflow/README.md]: Improved documentation

File: airflow/pyspark/dags/config/submit_dataproc_template_serverless.ini (hunk 5 above)
Comment (NiloFreitas) [reply airflow/pyspark/dags/config/submit_dataproc_template_serverless.ini]: Added documentation on how to run without schedule

**Review by shashank-google — APPROVED** [review]

**Pull request merged**